syntax = "proto3";
package etrs.proto;

import "Mesh.proto";
import "SoundSource.proto";
import "TempAndHumi.proto";
import "BotCar.proto";
import "BotGripper.proto";
import "BotMotor.proto";
import "BotArm.proto";
import "BotLed.proto";
import "KinectMode.proto";

message DataMessage {
    enum Type {
        OTHER = 0;
        MESH = 1;
        EXIT_MESH = -1;
        SOUND_SOURCE = 2;
        BOT_CAR = 3;
        BOT_GRIPPER = 4;
        BOT_MOTOR = 5;
        BOT_ARM = 6;
        BOT_LED = 7;
        TEMP_AND_HUMI = 8;
        KINECT_MODE= 9;
    }
    Type type = 1;
    oneof data {
        Mesh mesh = 2;
        SoundSource sound_source = 3;
        BotCar bot_car = 4;
        BotGripper bot_gripper = 5;
        BotMotor bot_motor = 6;
        BotArm bot_arm = 7;
        BotLed bot_led = 8;
        TempAndHumi temp_and_humi = 9;
        KinectMode kinect_mode = 10;
    }
}